### PR TITLE
Metrics port collision

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Bugfix

### Description

1. Throw an exception on port collision with admin http port
2. Change the default metrics port to `9190` so we can continue using `9192` as our default example cluster port
    
If the admin port is enabled and any virtual cluster has declared that it wants to bind to that port (either with a shared or exclusive port), then `PortConflictDetector` throws an `IllegalStateException`.
    
If either the admin endpoint or the virtual cluster is binding to the <any> interface (ie `0.0.0.0`), we check if their ports collide.

If both the admin endpoint and the virtual cluster are binding to the same specific interface (not `0.0.0.0`), we check if their ports collide.

### Additional Context

Since the addition of the port-per-broker cluster binding mechanism the default metrics port of `9193` has been colliding with the port range used for example config and documentation, with the bootstrap on `localhost:9192` and therefore broker-0 will bind on `9193` and conflict.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [x] Update documentation
- [x] Reference relevant issue(s) and close them after merging
- [x] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java [hunk 1]
```diff
@@ -60,7 +62,9 @@ private record EventGroupConfig(String name, EventLoopGroup bossGroup, EventLoop
         public List<Future<?>> shutdownGracefully() {
             return List.of(bossGroup.shutdownGracefully(), workerGroup.shutdownGracefully());
         }
-    };
+    }
```
Comment (robobario) [inline kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java:65]: weird formatting

File: kroxylicious/src/main/java/io/kroxylicious/proxy/Kroxylicious.java [hunk 2]
```diff
@@ -43,10 +43,10 @@ public Integer call() throws Exception {
 
         try (InputStream stream = Files.newInputStream(configFile.toPath())) {
             Configuration config = new ConfigParser().parseConfiguration(stream);
-
-            KafkaProxy kafkaProxy = new KafkaProxy(config);
-            kafkaProxy.startup();
-            kafkaProxy.block();
+            try (KafkaProxy kafkaProxy = new KafkaProxy(config)) {
```
Comment (robobario) [inline kroxylicious/src/main/java/io/kroxylicious/proxy/Kroxylicious.java:46]: unrelated tidy, intellij warned it wasn't in a try-with-resources

**Comment by robobario** [comment]
Now if I set:

```yaml
adminHttp:
  port: 9193
  endpoints:
    prometheus: {}
```

And run `java -jar kroxylicious/target/kroxylicious-0.3.0-SNAPSHOT.jar -c kroxylicious/example-proxy-config.yml`, the process exits immediately with:

```
java.lang.IllegalStateException: The exclusive bind of port(s) 9193 for virtual cluster 'demo' to <any> would conflict with another (non-cluster) port binding
	at io.kroxylicious.proxy.internal.PortConflictDetector.checkForConflictsWithOtherExclusivePort(PortConflictDetector.java:154)
	at io.kroxylicious.proxy.internal.PortConflictDetector.lambda$validate$13(PortConflictDetector.java:57)
```

**Review by SamBarker — APPROVED** [review]
LGTM

**Comment by robobario** [comment]
Thanks for review @SamBarker, will leave it for others to review since it can be worked around by removing the metrics config, changing the metrics port or changing the cluster bootstrap port

**Review by franvila — APPROVED** [review]
LGTM

**Review by k-wall — APPROVED** [review]

**Pull request merged**